xclim/indices/_agro.py

```diff
@@ -1067,17 +1069,186 @@ def _get_rain_season(pram):
+
+    def wrap_fit(da):
+        if indexer != {} and da.isnull().all():
+            select_dims = {d: 0 if d != "time" else [0, 1] for d in da.dims}
+            fitted = fit(da[select_dims], dist, method)
+            return fitted.broadcast_like(da.isel(time=0, drop=True))
+        return fit(da, dist, method)
+
+    cal_range = (
+        da.time.min().dt.strftime("%Y-%m-%d"),
+        da.time.max().dt.strftime("%Y-%m-%d"),
+    )
+    params = da.groupby(group).map(wrap_fit)
+    params.attrs = {
+        "Calibration period": cal_range,
+        "freq": freq,
+        "window": window,
+        "scipy_dist": dist,
+        "method": method,
+        "group": group,
+        "time_indexer": indexer,
+        "units": "",
+        "offset": offset,
+    }
+    return params
+
+
+def _get_standardized_index(da, params, **indexer):
+    scipy_dist = get_dist(params.attrs["scipy_dist"])
+    group_name = params.attrs["group"].rsplit(".")[-1]
+    param_names = params.dparams.values
+
+    def wrap_cdf_ppf(da, params, group_idxs):
+        if indexer != {} and np.isnan(da).all():
+            return np.zeros_like(da) * np.NaN
+
+        spi = np.zeros_like(da)
+        # loop over groups (similar to np.groupies idea, but np.groupies can't be used with custom functions)
+        for ii, group_idx in enumerate(list(dict.fromkeys(group_idxs).keys())):
+            pars = params[ii, :]
+            indices = np.argwhere(group_idxs == group_idx)
+            vals = da[indices]
+            zeros = (vals == 0).sum(axis=0)
+            vals[vals == 0] = np.NaN
+            paramsd = {param_names[jj]: pars[jj] for jj in range(len(pars))}
+            probs_of_zero = zeros / vals.shape[0]
+            dist_probs = scipy_dist.cdf(vals, **paramsd)
+            probs = probs_of_zero + ((1 - probs_of_zero) * dist_probs)
+            spi[indices] = norm.ppf(probs)
+        return spi
+
+    group_idxs = da.time.dt.month if group_name == "month" else da.time.dt.dayofyear
+    return xarray.apply_ufunc(
+        wrap_cdf_ppf,
+        da,
+        params,
+        group_idxs,
+        input_core_dims=[["time"], [group_name, "dparams"], ["time"]],
+        output_core_dims=[["time"]],
+        dask="parallelized",
+        dask_gufunc_kwargs={"allow_rechunk": True},
+        output_dtypes=[da.dtype],
+        vectorize=True,
+    )
+
+
+# TODO : ADD
+"""
+streamflow index (SSI)     GEV, log-logistic (Vincente-Serrano et al., 2012) Tweedie (Barker et al., 2016)    (Station de suivi)
+groundwater index (SGI)   log-normal, gamma, GEV (Bloomfield et Marchant, 2013)	                            (Station de suivi)
+NEW dist: GEV, log-normal
+https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.genextreme.html#scipy.stats.genextreme
+https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.lognorm.html
+"""
+
+
 @declare_units(
     pr="[precipitation]",
     pr_cal="[precipitation]",
+    params="[]",
```

Comment:
Is this OK? 

`pr_cal` is maintained temporary as a xr.DataArray | None
`params` is Quantified without units, does this need a declare_units?